### PR TITLE
Integrate Dart Formatter directly into ts2dart for cleaner output

### PR DESCRIPTION
**Pull request by jacob314**
and more readable test cases.

Not sure if the way I integrated dart-style.d.ts is the right but it works.

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -37,7 +38,10 @@ var tsProject =
 gulp.task('compile', function() {
   hasError = false;
   var tsResult =
-      gulp.src(['lib/**/*.ts', 'typings/**/*.d.ts', 'node_modules/typescript/lib/typescript.d.ts'])
+      gulp.src([
+            'lib/**/*.ts', 'typings/**/*.d.ts', 'node_modules/typescript/lib/typescript.d.ts',
+            'node_modules/dart-style/dart-style.d.ts'
```
Comment (mprobst) [inline gulpfile.js]: this should not be necessary, TypeScript should find this just from the `require('dart-style');`.

File: lib/formatter.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import style = require('dart-style');
+
+interface FormatResult {
```
Comment (mprobst) [inline lib/formatter.ts]: Shouldn't this interface live in the dart-style .d.ts file?

File: lib/formatter.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  code?: string;
+  error?: string;
+}
+
+/**
+ * Execute the Dart Formatter https://github.com/dart-lang/dart_style on either
+ * a single statement or an entire Dart library.
+ * Errors are logged if code cannot be properly formatted.
```
Comment (mprobst) [inline lib/formatter.ts]: Please fail, explosively? Silent failure is a total pain to debug later on.

File: lib/formatter.ts (hunk 2 above)
Comment (jacob314) [inline lib/formatter.ts]: removed

File: lib/formatter.ts (hunk 3 above)
Comment (jacob314) [inline lib/formatter.ts]: Moved this logic to the Transformer class and had output just return the unformatted string so I could use the existing logic to fail explosively when there was an error.

**Comment by jacob314** [comment]
ptal

File: lib/main.ts [hunk 4]
```diff
@@ -14,6 +14,7 @@ import StatementTranspiler from './statement';
 import TypeTranspiler from './type';
 import LiteralTranspiler from './literal';
 import {FacadeConverter} from './facade_converter';
+import * as style from 'dart-style';
```
Comment (mprobst) [inline lib/main.ts]: nit: call it `dartStyle` or so?

**Comment by mprobst** [comment]
Looks pretty good. Could you check out my other comment above?

Also: in this project's `package.json`, you should now have a dependency on `dart-style` - did you run `npm install dart-style --save`?

File: gulpfile.js (hunk 1 above)
Comment (jacob314) [inline gulpfile.js]: Fixed the module

**Comment by mprobst** [comment]
Merged as c43031f.

**Pull request closed without merging**